### PR TITLE
Repair fsp tests: Install xcp-ng-xapi-storage

### PR DESCRIPTION
**Pull request by Wescoeur**
Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

### CODE REVIEW

File: tests/storage/fsp/conftest.py [hunk 1]
```diff
@@ -1,24 +1,55 @@
 import pytest
 
+FSP_REPO_FILE = '/etc/yum.repos.d/xcp-ng-runx.repo'
+
+FSP_REPO_TEXT = """\
+[xcp-ng-runx]
+name=XCP-ng runx Repository
+baseurl=http://mirrors.xcp-ng.org/8/8.2/runx/x86_64/ http://updates.xcp-ng.org/8/8.2/runx/x86_64/
```
Comment (stormi) [inline tests/storage/fsp/conftest.py:8]: The URL should depend on the release of XCP-ng of the host being tested.

File: tests/storage/fsp/conftest.py [hunk 2]
```diff
@@ -1,24 +1,55 @@
...
+FSP_PACKAGES = ['xcp-ng-xapi-storage']
+
 DIRECTORIES_PATH = 'directories'
 
+def install_fsp(host):
+    assert not host.file_exists('/usr/bin/runx'), \
+        "runx should not be installed on the host before all tests"
+    host.create_file(FSP_REPO_FILE, FSP_REPO_TEXT)
+    host.yum_save_state()
+    host.yum_install(FSP_PACKAGES, enablerepo='epel')
```
Comment (stormi) [inline tests/storage/fsp/conftest.py:24]: 1. Enabling epel is probably not necessary if we don't install runx.
2. this install will work, provided the release of the xcp-ng-xapi-storage in the runx repo is always higher than that in xcp-ng-base and xcp-ng-updates repos. If you want to ensure this works (provided we don't install xcp-ng-xapi-storage on hosts by default someday), you should disable the other repos before installing.

File: tests/storage/fsp/test_fsp_sr.py [hunk 3]
```diff
@@ -36,17 +36,17 @@ def test_create_and_destroy_sr_non_existing_path(self, host):
 
 @pytest.mark.usefixtures("host_at_least_8_3")
```
Comment (stormi) [inline tests/storage/fsp/test_fsp_sr.py:37]: We can remove this restriction now that the runx repo is added to make it work.

File: tests/storage/fsp/conftest.py [hunk 4]
```diff
@@ -1,24 +1,55 @@
...
+"""
+
+FSP_PACKAGES = ['xcp-ng-xapi-storage']
+
 DIRECTORIES_PATH = 'directories'
 
+def install_fsp(host):
+    assert not host.file_exists('/usr/bin/runx'), \
+        "runx should not be installed on the host before all tests"
+    host.create_file(FSP_REPO_FILE, FSP_REPO_TEXT)
```
Comment (stormi) [inline tests/storage/fsp/conftest.py:22]: In addition to `create_file`, a Host method to add an optional xcp-ng yum repository (runx, linstor, ...) would be nice.

Example signature :

```
    def add_xcpng_repo(name):
        # creates the repo file for repo `name` from XCP-ng mirror
```

File: lib/host.py [hunk 5]
```diff
@@ -85,6 +85,31 @@ def stringify(key, value):
...
+        with tempfile.NamedTemporaryFile('w') as file:
+            file.write(text)
+            file.flush()
+            self.scp(file.name, filename)
+
+    def add_xcpng_repo(self, name):
+        content = """\
+[xcp-ng-{}]
+name=XCP-ng {} Repository
+baseurl=http://mirrors.xcp-ng.org/8/{}/{}/x86_64/ http://updates.xcp-ng.org/8/{}/{}/x86_64/
```
Comment (stormi) [inline lib/host.py:98]: It would not be a lot more work to replace "8" with the first number from the release.

File: lib/host.py [hunk 6]
```diff
@@ -85,6 +85,32 @@ def stringify(key, value):
...
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-xcpng
+"""
+
+        major = self.xcp_version.major
+        self.create_file(
+            '/etc/yum.repos.d/xcp-ng-{}.repo'.format(name),
+            content.format(name, name, major, self.xcp_version, name, major, self.xcp_version, name)
+        )
```
Comment (stormi) [inline lib/host.py:109]: Not a blocking comment, but I would find it much more readable with a f-string.

Comment (Wescoeur) [reply lib/host.py:109]: Right. I forget it's not python2. :)

**Review by stormi — APPROVED** [review]
Approving, to the condition that it was tested on both 8.2 and 8.3.

**Comment by stormi** [comment]
Is it just missing a review from @MatiasVara?

**Review by MatiasVara — APPROVED** [review]

**Pull request merged**